### PR TITLE
986/ingest videos

### PR DESCRIPTION
**Pull request by namrathaurs**
PR for ingestion of ACL2020 videos (#986). Also includes fixes made to incorporate @mjpost's general comments about the script in general. 

Files:
- `bin/add_video_tags.py` 
- `data/xml/<those_for_acl2020>`

### CODE REVIEW

**Comment by akoehn** [comment]
The `</paper>` should also be in its own line.

File: bin/add_video_tags.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        tree.write(f, encoding="UTF-8", xml_declaration=True)
+
+
+def main(args):
+    combo_df = combine_tsv(args['tsv_files'])
+    combo_df_uniques = combo_df['anthology_id'].apply(split_anth_id).unique()
+
+    for xml in os.listdir(data_dir):
+        fname, ext = os.path.splitext(xml)
+        if fname in combo_df_uniques.tolist() or fname == "2020.acl":
```
Comment (mjpost) [inline bin/add_video_tags.py:69]: Currently, you are opening, parsing, and writing the XML file for every paper. I suggest opening the XML file here, getting the `paper` node for each paper, and passing it into `add_video_tag()`. It should run much faster then.

File: bin/add_video_tags.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    coll_id, vol_id, paper_id = deconstruct_anthology_id(anth_paper.anthology_id)
+    tree = ET.parse(os.path.join(data_dir, xml_file))
+
+    paper = tree.getroot().find(f'./volume[@id="{vol_id}"]/paper[@id="{paper_id}"]')
+    video_url = "http://slideslive.com/{}".format(anth_paper.presentation_id)
+    make_simple_element("video", attrib={"tag": "video", "href": video_url}, parent=paper)
+
+    with open(os.path.join(data_dir, coll_id + ".xml"), 'wb') as f:
+        tree.write(f, encoding="UTF-8", xml_declaration=True)
+
```
Comment (mjpost) [inline bin/add_video_tags.py:61]: This should be moved into mail inside the `for xml in...` loop. And if you put the line

```
    anthology.utils.indent(tree.getroot())
```

just before the call to `tree.write()`, it should fix your indentation problem!

**Review by mjpost — COMMENTED** [review]
This looks awesome—thank you for all of this work! Just one small problem that should be easy to fix, and we'll be ready to merge.

**Comment by mjpost** [comment]
It looks like you've fallen behind the current `master` branch—can you merge that in? You might want to revert to the master branch version of `data/2020.acl.xml` first, to make the merge easier, and then reapply your video tags (with the above fixes).

File: bin/add_video_tags.py (hunk 1 above)
Comment (namrathaurs) [reply bin/add_video_tags.py:68]: This suggestion has been incorporated in commit 01588a9.

File: bin/add_video_tags.py (hunk 2 above)
Comment (namrathaurs) [reply bin/add_video_tags.py]: This suggestion has been incorporated in commit 01588a9.

**Comment by mjpost** [comment]
It looks like you need to do two things:

1. The build fails if your Python isn't formatted according [black guidelines](https://black.readthedocs.io/en/stable/). Run `make autofix` and then commit the changes to fix that.

2. You need to merge in master. You can do: 

```bash
git checkout master
git pull git@github.com:acl-org/acl-anthology master
git checkout [your branch]
git merge master
```

This might create a lot of merge conflicts since the master branch has diverged. In that case, it will likely be easier to *revert* to the master branch XML files, reapply the video tags, and commit.

**Review by mjpost — APPROVED** [review]
LGTM.

**Comment by namrathaurs** [comment]
Here's the tarball containing all the intermediate scripts and TSV files that was used for this PR - [intermediate_files_aclanth_pr999.tar.gz](https://github.com/acl-org/acl-anthology/files/5250861/intermediate_files_aclanth_pr999.tar.gz)

**Pull request merged**